### ISSUE TITLE
Ability to skip GOG Galaxy when launching game?

### ISSUE THREAD

**Issue by acheronfail** [moraroy/NonSteamLaunchers-On-Steam-Deck#492]
Apologies if this has already been asked before, but I couldn’t find it.

When the NSL game runs and imports GOG games, the shortcuts launch GOG Galaxy and then launch the game. This is nice for cloud saves and such, but it does add quite a bit of delay…

Could we somehow make an option to have these imported GOG games skip the launcher when running them?

**Comment by moraroy** [comment]
@acheronfail what you're looking for I think may be "Minimize to tray" it's in the Gog launcher settings.

**Comment by acheronfail** [comment]
Sorry, I probably wasn't clear enough in the original description!

I want the shortcuts created by NonSteamLaunchers to _launch the game itself_ and not launch GOG Galaxy at all - i.e., bypass GOG Galaxy completely.

For example, the game Dicey Dungeons on GOG should have its steam shortcut's target set to:

```
"/home/deck/.local/share/Steam/steamapps/compatdata/NonSteamLaunchers/pfx/drive_c/Program Files (x86)/GOG Galaxy/Games/Dicey Dungeons/diceydungeons.exe"
```

But right now, its target is:

```
"/home/deck/.local/share/Steam/steamapps/compatdata/NonSteamLaunchers/pfx/drive_c/Program Files (x86)/GOG Galaxy/GalaxyClient.exe"
```

(And the Launch Options tell GOG Galaxy to launch it.)

**Comment by moraroy** [comment]
@acheronfail how would you get your cloud saves and achievements etc if i do it that way? Also doing that option that I was talking about actually removes some "delay"

**Comment by acheronfail** [comment]
> how would you get your cloud saves and achievements etc if i do it that way?

Yes, I did mention this in the original message... For me, I'm okay without those things! I can always just launch GOG Galaxy itself and resync the cloud saves if would like to.

I suspect for users like myself where the Steam Deck is the only device I have for gaming, syncing between devices isn't a huge priority, and I'm happy for it to be a manual process. This is why I suggested that it could be an option, rather than the default.

When I see the GOG games in Steam, I'd like to be able to just launch them immediately without also having to launch GOG Galaxy each time, it adds quite a bit of delay since it has to launch from the start each time and even with "Minimise to tray" it still has to wait for GOG Galaxy to finish launching before it can launch the game itself.

**Comment by moraroy** [comment]
@acheronfail I understand, but I'm sorry brother, there may be a combination of certain settings in the gog launcher that may enable you to somewhat do this, but you would have to test all the combination of those yourself, on top of which if I were to just change it to opening the .exe I have to think about other people that wouldn't want that, especially people that are just using the plugin, they wouldn't want to open up gog everything just to "sync". Hopefully that makes sense. Currently on mine, I've actually switched to minimize to tray, as that seemed to remove some delay like you were saying. I'll look at some files and configure to see if I can find any more settings to get the delay down more. The point of NSL is to make sure you get all of the features of the launcher without you having to worry about things too much haha

**Comment by acheronfail** [comment]
Okay I understand. 👍 

For the time being I've removed GOG from NSL on my deck, and I've just manually moved the exe location mentioned in "Launch Options" into "Target", and each game boots immediately without any issues.

If you don't want to support optional settings in NSL I get that, too. Thanks for hearing me out!

**Comment by moraroy** [comment]
@acheronfail no worries, I'll leave the issue  open in case I find another way to do it like you were saying, also just FYI, on launchers like itchio and legacy I'm using the .exe directly. For launchers like these I havnt found a way yet to launch through the launcherbso the .exe is used directly.

**Comment by moraroy** [comment]
@acheronfail there is a check box to not sync to cloud in the settings just fyi, it just straight launches the game, even less delay time now

**Comment by moraroy** [comment]
@acheronfail I found a way!

**Comment by Karrq** [comment]
Hey @moraroy I was also interested in this feature, precisely for the same reason - I'll try disabling sync to cloud & minimize to tray on launch, but what would be the way you last mentioned? 
Is it something inherent to NSL or a set of settings?

**Comment by moraroy** [comment]
@Karrq both actually, I'm working on some new stuff, but if you were to go inside of the GOG Galaxy Launcher settings. You can edit and change as much options as you can set them all to go to the tray and you can even "uncheck" cloud saves. Feel free to explore all of the settings inside the GOG Launcher and test for yourself which ones work best for you. As for the new stuff I'm working, I'll try to update as soon as I can.

**Comment by moraroy** [comment]
@Karrq will also be adding this to the notes for GOG as i think it would be helpful, now that gog is finally fixed for now, i can continu to add this functinality in that i was working on
